src/components/table/mobile/responsive_context.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  useIsWithinMinBreakpoint,
+  type EuiBreakpointSize,
+} from '../../../services';
+import { useComponentDefaults } from '../../provider/component_defaults';
+
+export const DEFAULT_TABLE_BREAKPOINT: EuiBreakpointSize = 'm';
+
+/**
+ * Used by parent/top-level table components to determine isResponsive state
+ * based on the passed breakpoint
+ */
+export const useIsEuiTableResponsive = (
+  componentProp?: EuiBreakpointSize | boolean
+): boolean => {
+  const componentDefault =
+    useComponentDefaults().EuiTable?.responsiveBreakpoint;
+  const breakpoint =
+    componentProp ?? componentDefault ?? DEFAULT_TABLE_BREAKPOINT;
+
+  const isBoolean = typeof breakpoint === 'boolean';
+
+  // We use ! and minBreakpoint to more accurately reflect the single point at which tables collapse
+  const isResponsive = !useIsWithinMinBreakpoint(isBoolean ? '' : breakpoint);
```

Comment:
This might just be me, but I had to read these two lines more than once to get what is what.

Would this change make sense to simplify?
```suggestion
return isBoolean ? breakpoint : !useIsWithinMinBreakpoint(breakpoint);
```

Reply:
Sadly hooks can't be called conditionally, otherwise I'd definitely do that 🥲 

Also I think my comment is being confusing - it's meant to address why I didn't use `useIsWithinMaxBreakpoint` (as opposed to `!useIsWithinMinBreakpoint` - and the answer to that is that these are equivalent:

```
useIsWithinMaxBreakpoint('s')
!useIsWithinMinBreakpoint('m')
```

But I wanted the 2nd one because the `m` breakpoint is `768`px and I wanted it to be clear that below that is where the table will break down responsively. (tbh in general I think our breakpoint utilities are named super confusingly but maybe that's just me 😅)

Reply:
>Sadly hooks can't be called conditionally

Ah shoot, yeah I totally ignored that 🙈 Yeah then your implementation is totally fine.

And thanks for the extra context for choosing the helper! I did understand the general idea from the comment but yeah, it's a bit hard to read with the `!` _and_ overall abstract helper name. I had to check on the helpers in more depth.

Reply:
I've updated the comment a bit more to be less vague and more specific! Hopefully that helps, let me know if you have any further suggestions! https://github.com/elastic/eui/pull/7625/commits/edaf2d3b113805165ce30be2ce5fe1981beeecda